mollie/refunds.go

```diff
@@ -0,0 +1,243 @@
+	Refunded   RefundStatus = "refunded"
+	Failed     RefundStatus = "failed"
+)
+
+// OrderLines describes an array of order line objects.
+type OrderLines struct {
+	Quantity       int     `json:"quantity,omitempty"`
+	DiscountAmount *Amount `json:"discountAmount,omitempty"`
+	VatAmount      *Amount `json:"vatAmount,omitempty"`
+	TotalAmount    *Amount `json:"totalAmount,omitempty"`
+}
+
+// RefundLinks describes all the possible links to be returned with
+// a Refund object.
+type RefundLinks struct {
+	Self       *URL `json:"self,omitempty"`
+	Payment    *URL `json:"payment,omitempty"`
+	Settlement *URL `json:"settlement,omitempty"`
+	Order      *URL `json:"order,omitempty"`
+}
+
+// RefundOptions describes refund endpoint valid query string parameters.
+//
+// See: https://docs.mollie.com/reference/v2/refunds-api/get-refund.
+type RefundOptions struct {
+	Embed    EmbedValue `url:"embed,omitempty"`
+	TestMode bool       `url:"testmode,omitempty"`
+}
+
+// EmbedValue describes the valid value of embed query string.
+type EmbedValue string
+
+// Valid Embed query string value.
+const (
+	EmbedPayment EmbedValue = "payment"
+)
+
+// ListRefundOptions describes list refund endpoint valid query string parameters.
+//
+// See: https://docs.mollie.com/reference/v2/refunds-api/list-refunds.
+type ListRefundOptions struct {
+	From      string     `url:"from,omitempty"`
+	Limit     string     `url:"limit,omitempty"`
+	ProfileID string     `url:"profileId,omitempty"`
+	TestMode  bool       `url:"testmode,omitempty"`
+	Embed     EmbedValue `url:"embed,omitempty"`
+}
+
+// RefundsService instance operates over refund resources.
+type RefundsService service
+
+// Get retrieve a single refund by its ID.
+//
+// If you do not know the original payment’s ID, you can use the List payment refunds endpoint.
+func (rs *RefundsService) Get(paymentID, refundID string, options *RefundOptions) (refund Refund, err error) {
+	u := fmt.Sprintf("v2/payments/%s/refunds/%s", paymentID, refundID)
+	if options != nil {
+		v, _ := query.Values(options)
+		u = fmt.Sprintf("%s?%s", u, v.Encode())
+	}
+
+	req, err := rs.client.NewAPIRequest(http.MethodGet, u, options)
+	if err != nil {
+		return
+	}
+
+	res, err := rs.client.Do(req)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(res.content, &refund); err != nil {
+		return
+	}
+
+	return
+}
+
+var (
+	requiredCreateParamRefund = "parameter required for creating a refund: %+v"
+)
+
+// Create a refund payment request.
+//
+// See https://docs.mollie.com/reference/v2/refunds-api/create-refund.
+func (rs *RefundsService) Create(paymentID string, re Refund, options *RefundOptions) (rf Refund, err error) {
+	if re.Amount == nil {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount)
+	}
+
+	if re.Amount.Currency == "" {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount.Currency)
+	}
+
+	if re.Amount.Value == "" {
+		return re, fmt.Errorf(requiredCreateParamRefund, re.Amount.Value)
+	}
```

Comment:
What do you think about moving this to an unexported function in this same file? 
It will improve readability and reduce the complexity of this method.

Reply:
Sounds good to me. I'll implement it.